tools/online_editor/src/index.ts

```diff
@@ -209,21 +218,16 @@ export Demo := Window {
     if (new_current != undefined) {
       new_current.className = "nav-link active";
     }
-    const model_and_state = editor_documents.get(url);
-    if (model_and_state != undefined) {
-      editor.setModel(model_and_state.model);
-      if (model_and_state.view_state != null) {
-        editor.restoreViewState(model_and_state.view_state);
-      }
-      editor.focus();
+    const model_and_state = editor_documents.get(url)!;
+    editor.setModel(model_and_state.model);
+    if (model_and_state.view_state != null) {
+      editor.restoreViewState(model_and_state.view_state);
     }
+    editor.focus();
   }
 
   function update_preview() {
-    const main_model_and_state = editor_documents.get(base_url);
-    if (main_model_and_state === undefined) {
-      return;
-    }
+    const main_model_and_state = editor_documents.get(base_url)!;
```

Comment:
is it valid to use the `!`? (here and everywhere else)
Is this really never undefined? What happens when it is?

Reply:
It shouldn't be: The code should not be reached if this is invalid.

The code will throw an exception and stop if this is triggered.